tools/manifest/manifest.py

```diff
@@ -229,9 +229,14 @@ def load(tests_root, manifest):
 
 
 def write(manifest, manifest_path):
+    from tempfile import NamedTemporaryFile
+
     dir_name = os.path.dirname(manifest_path)
     if not os.path.exists(dir_name):
         os.makedirs(dir_name)
-    with open(manifest_path, "wb") as f:
-        json.dump(manifest.to_json(), f, sort_keys=True, indent=1, separators=(',', ': '))
-        f.write("\n")
+
+    temp_manifest = NamedTemporaryFile(mode="wb", dir=dir_name)
```

Comment:
I think we should actually use `mkstemp` directly since the semantcs of `NamedTemporaryFile` are to delete the file, which isn't going to work in this case since it's renamed.

Reply:
Yes it does give a silent warning when it attempts to delete the file with NamedTemporaryFile, so I'll change that.